library/ssl_cli.c

```diff
@@ -1363,7 +1363,7 @@ static int ssl_parse_ecjpake_kkpp( mbedtls_ssl_context *ssl,
 {
     int ret;
 
-    if( ssl->transform_negotiate->ciphersuite_info->key_exchange !=
+    if( ssl->handshake->ciphersuite_info->key_exchange !=
```

Comment:
Note (again no change requested here, but for the code size reduction work): I wonder if we should have a smart function for testing this that would compile to a constant when only one key exchange is compiled in.

Reply:
Yes, that's a little bit similar to removing version and protocol checks entirely if we know that there's only one enabled.